subprojects/internal-integ-testing/src/main/groovy/org/gradle/test/fixtures/maven/MavenPublishedJavaModule.groovy

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.test.fixtures.maven
+
+import org.gradle.test.fixtures.PublishedJavaModule
+
+class MavenPublishedJavaModule extends DelegatingMavenModule<MavenFileModule> implements PublishedJavaModule {
```

Comment:
I'm still confused a bit by the name of this class, where it actually consists of publishing a "Gradle Module", but have no better suggestion at this point.

Reply:
The `MavenModule` has 2 purposes: to generate a module for testing, and to verify a module that was separately published. This class is currently only used for verifying an existing publication.

But I guess we might later add the ability to do `mavenJavaModule.apiDependsOn("org:foo:1.0").runtimeDependsOn('org:bar:1.0').publish()`, which would generate the correct Gradle/POM metadata. So maybe I'll rename it to simply `MavenJavaModule`. Ideally we'd be switch this stuff to using traits instead of the nasty inheritance hierarchy.